`test_scale_run_script_deterministic` failing on `master` on Actions runner

Post merging in #1470 the `tests/test_determinism.py::test_scale_run_script_deterministic` test is [failing](https://github.com/UCL/TLOmodel/actions/runs/11158917558/job/31016247252) in runs of the test workflows on the GitHub Actions runners:

<details><summary><code>tests.test_determinism.test_scale_run_script_deterministic</code></summary>

<pre><code>tests/test_determinism.py:46: in test_scale_run_script_deterministic
    assert completed_process.returncode == 0, (
E   AssertionError: Running /mnt/tlo/6/_work/TLOmodel/TLOmodel/.tox/py311/bin/python /mnt/tlo/6/_work/TLOmodel/TLOmodel/src/scripts/profiling/scale_run.py --years 1 --months 6 --initial-population 1000 --seed 645407762 --output-dir /tmp/pytest-of-gha/pytest-12557/test_scale_run_script_determin0 --save-final-population fails to successfully complete with stderr output
E     
E     None
E   assert 1 == 0
E    +  where 1 = CompletedProcess(args=['/mnt/tlo/6/_work/TLOmodel/TLOmodel/.tox/py311/bin/python', '/mnt/tlo/6/_work/TLOmodel/TLOmodel/src/scripts/profiling/scale_run.py', '--years', '1', '--months', '6', '--initial-population', '1000', '--seed', '645407762', '--output-dir', '/tmp/pytest-of-gha/pytest-12557/test_scale_run_script_determin0', '--save-final-population'], returncode=1).returncode</code></pre></details>

This test is passing for me locally. I suspect this may be due to the fact that the [`psutil.Process.memory_full_info` method](https://psutil.readthedocs.io/en/latest/index.html#psutil.Process.memory_full_info) 'usually requires higher user privileges' to run compared to [`psutil.Process.memory_full_info`](https://psutil.readthedocs.io/en/latest/index.html#psutil.Process.memory_info), and potentially the process doesn't have sufficient privileges when run on an Actions runner?

We only detected this failure on merging to `master` as the tests workflow was skipped for the pull request as the `paths` filter ignores changes in `src/scripts`. As we call `src/scripts/profiling/scale_run.py` in the test we should add this file and the import `src/scripts/profiling/shared.py` to the `paths` list to avoid this happening in future!

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -10,6 +10,8 @@ on:
       - requirements/**
       - resources/**
       - src/tlo/**
+      - src/scripts/profiling/scale_run.py
+      - src/scripts/profiling/shared.py
       - tests/**
       - pyproject.toml
       - tox.ini

--- a/src/scripts/profiling/shared.py
+++ b/src/scripts/profiling/shared.py
@@ -23,16 +23,12 @@ def memory_statistics() -> dict[str, float]:
         Resident set size in mebibytes. The non-swapped physical memory the process has used.
     memory_vms_MiB: float
         Virtual memory size in mebibytes. The total amount of virtual memory used by the process.
-    memory_uss_MiB: float
-        Unique set size in mebibytes. The memory which is unique to a process and which would be freed if the process
-        was terminated right now
     """
     process = psutil.Process()
-    memory_info = process.memory_full_info()
+    memory_info = process.memory_info()
     return {
         "memory_rss_MiB": memory_info.rss / 2**20,
         "memory_vms_MiB": memory_info.vms / 2**20,
-        "memory_uss_MiB": memory_info.uss / 2**20,
     }
 
 

--- a/src/scripts/profiling/shared.py
+++ b/src/scripts/profiling/shared.py
@@ -23,12 +23,16 @@ def memory_statistics() -> dict[str, float]:
         Resident set size in mebibytes. The non-swapped physical memory the process has used.
     memory_vms_MiB: float
         Virtual memory size in mebibytes. The total amount of virtual memory used by the process.
+    memory_uss_MiB: float
+        Unique set size in mebibytes. The memory which is unique to a process and which would be freed if the process
+        was terminated right now
     """
     process = psutil.Process()
-    memory_info = process.memory_info()
+    memory_info = process.memory_full_info()
     return {
         "memory_rss_MiB": memory_info.rss / 2**20,
         "memory_vms_MiB": memory_info.vms / 2**20,
+        "memory_uss_MiB": memory_info.uss / 2**20,
     }
 
 

--- a/src/scripts/profiling/shared.py
+++ b/src/scripts/profiling/shared.py
@@ -3,7 +3,11 @@ import random
 import time
 
 import pandas as pd
-import psutil
+
+try:
+    import psutil
+except ImportError:
+    psutil = None
 
 from tlo import DateOffset, Simulation, logging
 from tlo.events import PopulationScopeEventMixin, RegularEvent
@@ -15,8 +19,8 @@ logger.setLevel(logging.INFO)
 
 def memory_statistics() -> dict[str, float]:
     """
-    Extract memory usage statistics in current process using `psutil`.
-    Statistics are returned as a dictionary.
+    Extract memory usage statistics in current process using `psutil` if available.
+    Statistics are returned as a dictionary. If `psutil` not installed an empty dict is returned.
     
     Key / value pairs are:
     memory_rss_MiB: float
@@ -27,6 +31,8 @@ def memory_statistics() -> dict[str, float]:
         Unique set size in mebibytes. The memory which is unique to a process and which would be freed if the process
         was terminated right now
     """
+    if psutil is None:
+        return {}
     process = psutil.Process()
     memory_info = process.memory_full_info()
     return {

